features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Subscriptions/SubscriptionsTable.jsx

```diff
@@ -27,8 +38,162 @@ const styles = theme => ({
+            />
+        </TableCell>
+        <TableCell>
+            <FormattedMessage
+                id='Apis.Details.Subscriptions.SubscriptionsTable.tier'
+                defaultMessage='Tier'
+            />
+        </TableCell>
+        <TableCell>
+            <FormattedMessage
+                id='Apis.Details.Subscriptions.SubscriptionsTable.status'
+                defaultMessage='Status'
+            />
+        </TableCell>
+        <TableCell>
+            <ScopeValidation
+                resourceMethod={resourceMethod.POST}
+                resourcePath={resourcePath.BLOCK_SUBSCRIPTION}
+            >
+                <FormattedMessage
+                    id='Apis.Details.Subscriptions.SubscriptionsTable.actions'
+                    defaultMessage='Actions'
+                />
+            </ScopeValidation>
+        </TableCell>
+    </TableRow>
+);
+
+const subscriptionStatus = {
+    BLOCKED: 'BLOCKED',
+    PROD_BLOCKED: 'PROD_ONLY_BLOCKED',
+};
+
+/**
+ * Table pagination for subscriptions table
+ *
+ * @param props props used for SubscriptionTablePagination
+ * @returns {*}
+ */
+function SubscriptionTablePagination(props) {
+    // enable allowShortCircuit as specified in https://eslint.org/docs/rules/no-unused-expressions
+    /* eslint no-unused-expressions: ["error", { "allowShortCircuit": true }] */
+    const {
+        count, page, rowsPerPage, onChangePage,
+    } = props;
+
+    /**
+     * handleFirstPageButtonClick loads data of the first page
+     * */
+    function handleFirstPageButtonClick() {
+        onChangePage && onChangePage(0);
+    }
+
+    /**
+     * handleBackButtonClick load data of the prev page
+     * */
+    function handleBackButtonClick() {
+        onChangePage && onChangePage(page - 1);
+    }
+
+    /**
+     * handleNextButtonClick load data of the next page
+     * */
+    function handleNextButtonClick() {
+        onChangePage && onChangePage(page + 1);
+    }
+
+    /**
+     * handleLastPageButtonClick load data of the last page
+     * */
+    function handleLastPageButtonClick() {
+        onChangePage && onChangePage(Math.max(0, Math.ceil(count / rowsPerPage) - 1));
+    }
+
+    return (
+        <div
+            style={{ display: 'flex' }}
+        >
+            <IconButton
+                onClick={() => handleFirstPageButtonClick()}
+                disabled={page === 0}
+            >
+                <FirstPageIcon />
+            </IconButton>
+            <IconButton
+                onClick={() => handleBackButtonClick()}
+                disabled={page === 0}
+            >
+                <KeyboardArrowLeft />
+            </IconButton>
+            <IconButton
+                onClick={() => handleNextButtonClick()}
+                disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+            >
+                <KeyboardArrowRight />
+            </IconButton>
+            <IconButton
+                onClick={() => handleLastPageButtonClick()}
```

Comment:
I think we could simply use

```suggestion
                onClick={handleLastPageButtonClick}
```

here and above places, no need to wrap it with a lambda function??